### ISSUE TITLE
Memory leak in Tree/TreeItem after running complex workflow

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-ui#227]
See #222 and #225 for background.

For some strange reason the workflow `five` doesn't leave anything behind. But the `complex`, after running ~12 minutes left a few `HTMLDivElement` detached in the heap.

![Screenshot_2019-09-16_14-43-51](https://user-images.githubusercontent.com/304786/64932153-330f2980-d891-11e9-9a79-abe62a3c80b7.png)

**Comment by kinow** [comment]
I created a branch and cherry-picked the commit from #251 PR. Then tested the workflow `five`.

![Screenshot_2019-09-24_09-02-15](https://user-images.githubusercontent.com/304786/65465644-f060d900-deb0-11e9-9edf-3edf07bdccf2.png)
![Screenshot_2019-09-24_09-01-44](https://user-images.githubusercontent.com/304786/65465646-f060d900-deb0-11e9-9d23-1a74a875ed79.png)
![Screenshot_2019-09-24_09-00-22](https://user-images.githubusercontent.com/304786/65465647-f0f96f80-deb0-11e9-85a8-9a1c109b657e.png)

Memory stayed within 24-27 MB before/after the tests. Which is fine.

The issue is still with the `complex` workflow. It now gave me a relatively OK handsaw pattern, and initial memory of 184MB at 09:15AM.

![Screenshot_2019-09-24_09-14-58](https://user-images.githubusercontent.com/304786/65465754-2900b280-deb1-11e9-8a76-491aff515e5a.png)

Didn't touch the VM for 10 minutes, then at 09:25 AM decided to take another snapshot. The browser stopped the snapshot at 39% progress, and then the VM became unresponsive, and I had to restart it :roll_eyes: .

**Comment by kinow** [comment]
Not really a blocker for 0.2, so if necessary can be moved to 0.3 or 1.0 👍

**Comment by kinow** [comment]
This issue hasn't occurred to me in a long time. I used to get browser tabs crashing with complex, and browser used memory going over the rooftop. Looking at the memory heap in the browser I'd see things like near 1GB for Cylc UI tab after running multiple workflows in a day.

But for the past few months since we have been improving the tree view (exploring infinite-tree, deltas, etc) this hasn't happened.

Closing :tada:
